Week04/lesson08/homework/temp_converter/js/indexbonus3.js

```diff
@@ -0,0 +1,57 @@
+function conversion() {
+
+  var c = document.querySelector("#c").value;
+  var f = document.querySelector("#f").value;
+
+  if ((f === null || f === "") && (c === null || c === "")) {
+      document.querySelector("#alert").innerHTML = "Please fill one of the fields above.";
+  }
+
+  else if (f === null || f === "") {
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    c = c+"°C";
+  }
+
+  else if (c === null || c === "") {
+    c = (f - 32) * 5 / 9;
+    document.querySelector("#c").value = Math.round(c*10)/10+"°C";
+    f = f+"°F";
+  }
+
+  else {
+    document.querySelector("#alert").innerHTML = "Please only fill one of the fields above. Click 'Reset' to clear both fields.";
+  }
+
+}
+
+document.querySelector("#convert").onclick = conversion;
+
+
+
+
+function clear() {
+  document.querySelector("#c").value = "";
+  document.querySelector("#f").value = "";
+  document.querySelector("#alert").innerHTML = "";
+}
+
+document.querySelector("#reset").onclick = clear;
+
+
+
+
+function clearAlert() {
+      document.querySelector("#alert").innerHTML = "";
+}
+
+document.querySelector("#c").onfocus = clearAlert;
```

Comment:
Got you - I have cleaned a bit my clearAlert function now...

function clearAlert() {
      document.querySelector("#alert").innerHTML = "";
}

document.querySelector("#c").onfocus = clearAlert;
document.querySelector("#f").onfocus = clearAlert;

... although I still want to different Clear functions in this case: one that clears both fields (and any potential alert message displayed) when clicking the Reset button and one that only clears the alert message when the user focuses on any field if both fields have a value when the user tries to click the Convert button. The latter one is to prevent a user from putting in a value in each field and trying to click Convert for instance.